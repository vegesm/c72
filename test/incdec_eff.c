main() {
    extern iarr;
    int iarr[];
    int p[];
    int a,b;
    char c, d, carr[4], cp[];

    a = 4;
    a++;
    printf("test1: %d\n", a);

    --a;
    printf("test2: %d\n", a);

    c = 15; c++;
    printf("test3: %d\n", c);

    c = -128; ++c;
    printf("test4: %d\n", c);

    iarr[2]++;
    printf("test5: %d %d %d\n", iarr[1], iarr[2], iarr[3]);

    p = iarr;
    p++;
    printf("test6: %d\n", *p);

    carr[0] = 3; carr[1] = 10; carr[2] = -128;
    carr[0]++;
    printf("test7: %d %d\n", carr[0], carr[1]);

    cp = carr+1;
    cp++;
    printf("test8: %d\n", *cp);
}

iarr[] 1, 8, 10, 128;

