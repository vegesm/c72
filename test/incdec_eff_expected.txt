test1: 5
test2: 4
test3: 16
test4: -127
test5: 8 11 128
test6: 8
test7: 4 10
test8: -128
